File: oauth-authenticator-archetype/src/main/resources/archetype-resources/src/main/java/config/__pluginName__AuthenticatorPluginConfig.java
package ${package}.config;

import se.curity.identityserver.sdk.config.Configuration;
import se.curity.identityserver.sdk.config.annotation.DefaultString;
import se.curity.identityserver.sdk.config.annotation.DefaultURI;
import se.curity.identityserver.sdk.config.annotation.Description;
import se.curity.identityserver.sdk.service.SessionManager;

import java.net.URI;

@SuppressWarnings("InterfaceNeverImplemented")
public interface ${pluginName}AuthenticatorPluginConfig extends Configuration {

	@Description("Client id")
    String getClientId();

    @Description("Secret key")
    String getClientSecret();

    @Description("URL to the upstream OAuth server's authorization endpoint")
    @DefaultURI("https://identity.example.com/authorize")
    URI getAuthorizationEndpoint();

    @Description("URL to the upstream OAuth server's token endpoint")
    @DefaultURI("https://identity.example.com/token")
    URI getTokenEndpoint();

    @Description("URL to upstream OpenID Connect Provider's user info endpoint")
    @DefaultURI("https://identity.example.com/user-info")
    URI getUserInfoEndpoint();

    @Description("A space-separated list of scopes to request from Custom")
    @DefaultString("")
    String getScope();

    SessionManager getSessionManager();
}

File: kotlin-oauth-authenticator-archetype/src/main/resources/archetype-resources/src/main/kotlin/config/__pluginName__AuthenticatorPluginConfig.kt
package ${package}.config


import se.curity.identityserver.sdk.config.Configuration
import se.curity.identityserver.sdk.config.annotation.Description
import se.curity.identityserver.sdk.service.ExceptionFactory
import se.curity.identityserver.sdk.service.HttpClient
import se.curity.identityserver.sdk.service.Json
import se.curity.identityserver.sdk.service.SessionManager
import se.curity.identityserver.sdk.service.WebServiceClientFactory
import se.curity.identityserver.sdk.service.authentication.AuthenticatorInformationProvider

import java.util.Optional

interface ${pluginName}AuthenticatorPluginConfig: Configuration
{
    @Description("Client id")
    fun getClientId(): String

    @Description("Secret key")
    fun getClientSecret(): String

    @Description("The HTTP client with any proxy and TLS settings that will be used to connect to the provider")
    fun getHttpClient(): Optional<HttpClient>

    fun getSessionManager(): SessionManager

    fun getExceptionFactory(): ExceptionFactory

    fun getAuthenticatorInformationProvider(): AuthenticatorInformationProvider

    fun getWebServiceClientFactory(): WebServiceClientFactory

    fun getJson(): Json
}
